src/constructs/kinesis/kinesis-stream.ts

```diff
@@ -1,15 +1,13 @@
-import type { CfnStream, StreamProps } from "@aws-cdk/aws-kinesis";
+import type { StreamProps } from "@aws-cdk/aws-kinesis";
 import { Stream } from "@aws-cdk/aws-kinesis";
 import type { GuStack } from "../core";
+import { GuMigratingResource } from "../core/migrating";
 
-export interface GuKinesisStreamProps extends StreamProps {
-  overrideId?: boolean;
-}
+export interface GuKinesisStreamProps extends StreamProps, GuMigratingResource {}
```

Comment:
I think we might want to implement `GuStatefulConstruct` here because:

* Kinesis streams can retain data for X days
* Stack A might use a Kinesis stream as way to accept input from stack B

Consequently I think it'd be helpful to receive [this warning](https://github.com/guardian/cdk/blob/54502809de5a111f26ad1574b78d488a77e37b85/src/constructs/core/migrating.ts#L49-L53) if the migration props have not been configured correctly.